solc/CommandLineParser.h

```diff
@@ -178,19 +178,19 @@ struct CommandLineOptions
 
 	struct
 	{
-		boost::filesystem::path dir;
+		std::optional<boost::filesystem::path> dir;
 		bool overwriteFiles = false;
 		langutil::EVMVersion evmVersion;
 		bool experimentalViaIR = false;
-		RevertStrings revertStrings = RevertStrings::Default;
+		std::optional<RevertStrings> revertStrings;
 		std::optional<langutil::DebugInfoSelection> debugInfoSelection;
-		CompilerStack::State stopAfter = CompilerStack::State::CompilationSuccessful;
+		std::optional<CompilerStack::State> stopAfter;
 	} output;
 
-	struct
+	struct 
 	{
-		yul::AssemblyStack::Machine targetMachine = yul::AssemblyStack::Machine::EVM;
-		yul::AssemblyStack::Language inputLanguage = yul::AssemblyStack::Language::StrictAssembly;
+		std::optional<yul::AssemblyStack::Machine> targetMachine;
+		std::optional<yul::AssemblyStack::Language> inputLanguage;
```

Comment:
These must still be initialized if the respective CLI options are not used. Just only in assembly mode. For example currently `targetMachine` will be `nullopt` when `!m_args.count(g_strMachine)` because the code was relying on these defaults.